pkg/monitoring/top/top_nonwindows.go

```diff
@@ -23,21 +21,25 @@ func (t *Top) GetProcesses() ([]*ProcessInfo, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "Failed to get processes")
 	}
+
+	skipped := 0
 	// Fetch load percentage for every process
 	for _, p := range ps {
 		// Run in background because the call to Percent blocks for the duration
 		go func(p *process.Process) {
 			load, err := p.Percent(time.Second * 1)
-			if err != nil && !strings.Contains(err.Error(), "no such file") {
-				// Ignore the file not not found case to avoid a lot of error messages for
-				// short lived processes
-				log.Printf("Err getting Percent: %s", err)
+			if err != nil {
+				// If we log the error in this place, we get _a lot of_ messages
+				skipped++
```

Comment:
it's not a good idea to increment the counter inside goroutine inside goroutine. It is not safe.

It's better to use sync.WaitGroup and close the channel after

Reply:
> It is not safe

Incrementing is an atomic operation, there is no `if` with a following instruction or similar. Therefor it's ok to do this in a goroutine

Reply:
Unfortunately incrementing is not an atomic operation. That's why sync.atomic exists

Reply:
Addressed